timeline/templatetags/timeline_tags.py

```diff
@@ -85,9 +86,19 @@ def get_app_icon_html(event, plugin_lookup):
         plugin_name = event.plugin if event.plugin else event.app
         if plugin_name in plugin_lookup.keys():
             plugin = plugin_lookup[plugin_name]
+            if not isinstance(plugin, ProjectAppPluginPoint):
+                url_kwargs['project'] = None
```

Comment:
This looks like it should work fine :) However, there's still one small thing to refactor. Right now, you may first set the project UUID and then reset it to None (see line 76), which is a bit redundant.

If the app is projectroles, you can simply add the project as a kwarg directly to the `reverse()` call on line 82 and set up the `url_kwargs` variable later, if we find a plugin on line 88.

Reply:
In that way it is also possible to set up `url_kwargs` variable not on line 88, but later (in new version of file it is line 89) in another `reverse()` for `entry_point`.